calico_kubernetes/calico_kubernetes.py

```diff
@@ -182,16 +173,34 @@ def _configure_interface(self):
         4) Assign the node's IP to the host end of the veth pair (required for
            compatibility with kube-proxy REDIRECT iptables rules).
         """
+        logger.info('Configuring Calico network interface')
+
         # Set up parameters
         container_pid = self._get_container_pid(self.docker_id)
         interface = 'eth0'
+        endpoint = self._container_add(container_pid, interface)
+        namespace = netns.PidNamespace(container_pid)
 
+        # Delete the existing veth connecting to the docker bridge.
         self._delete_docker_interface()
-        logger.info('Configuring Calico network interface')
-        ep = self._container_add(container_pid, interface)
-        interface_name = generate_cali_interface_name(IF_PREFIX, ep.endpoint_id)
+
+        # Create the veth, move into the container namespace, add the IP and
+        # set up the default routes.
+        logger.info("Creating the veth with namespace pid %s on interface name %s", container_pid, interface)
+        endpoint.mac = endpoint.provision_veth(namespace, interface)
+
+        # Update the endpoint to set the new mac address
+        logger.info("Setting mac address %s to endpoint %s", endpoint.mac, endpoint.name)
+        self._datastore_client.set_endpoint(endpoint)
+
+        resource_path = "namespaces/%(namespace)s/pods/%(podname)s" % \
+                        {"namespace": self.namespace, "podname": self.pod_name}
+        ep_data = '{"metadata":{"annotations":{"%s":"%s"}}}' % \
+                  (EPID_ANNOTATION_KEY, endpoint.endpoint_id)
+        self._patch_api(path=resource_path, patch=ep_data)
+
+        interface_name = generate_cali_interface_name(IF_PREFIX, endpoint.endpoint_id)
```

Comment:
Isn't the endpoint name accessible from endpoint.name (or something like that?)

Do we really need to be generating it again here?